validation/src/test/scala/hmda/validation/rules/lar/validity/V355Spec.scala

```diff
@@ -0,0 +1,58 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.dsl.{ Failure, Success }
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.LarEditCheckSpec
+import org.scalacheck.Gen
+
+class V355Spec extends LarEditCheckSpec with BadValueUtils {
+
+  property("If agency code is valid, lar with valid denial reasons must pass") {
+    forAll(larGen) { lar =>
+      whenever(agencyCodeValid(lar)) {
+        lar.mustPass
+      }
+    }
+  }
+
+  property("If agency code is 1 and action taken is 3 or 7, lar with valid denial reasons must pass") {
+    forAll(larGen) { lar =>
+      val validLar = lar.copy(agencyCode = 1)
+      whenever(agencyCodeAndActionTakenValid(validLar)) {
+        validLar.mustPass
+      }
+    }
+  }
+
+  val invalidAgencyCode: Gen[Int] = Gen.oneOf(4, 6, 8)
+  property("If both agency code and action taken are invalid, lar should pass") {
+    forAll(larGen, invalidAgencyCode) { (lar: LoanApplicationRegister, x: Int) =>
+      val validLar = lar.copy(agencyCode = x)
+      whenever(!agencyCodeAndActionTakenValid(validLar)) {
```

Comment:
I have been creating a custom generator for these that I copy into things like actionTaken instead of filtering all possible with a whenever, which I find to be a bit prettier. As long as you get enough hits though its really only comes down to a stylistic difference.